File: examples/erlang/mega_flow/storage.erl
%% Storage — pattern-match on record, accessor fn, procedural
%% dispatch to the sink.
-module(storage).
-export([persist/1]).

-include("envelope.hrl").

cmd_of(#envelope{cmd = Cmd}) -> Cmd.

run(#envelope{} = Envelope) ->
    C = cmd_of(Envelope),
    executor:execute(C).

persist(#envelope{} = Envelope) ->
    run(Envelope).

File: examples/lua/mega_flow/storage.lua
-- Storage — module-return table pattern + procedural dispatch.
-- Taint rides through the accessor and wrapper into the sink.
local Executor = require("executor")
local M = {}

local function cmd_of(envelope)
    return envelope.cmd, envelope.user
end

function M.run(envelope)
    local cmd, _user = cmd_of(envelope)
    return Executor.execute(cmd)
end

function M.persist(envelope)
    return M.run(envelope)
end

return M
